Fails to build on macOS 10.14.6/Xcode 10.2.1

```
$ make -j4
[ 11%] Building CXX object CMakeFiles/jiyu.dir/src/os_support.cpp.o
jiyu/src/os_support.cpp:47:19: error: use of undeclared identifier '_NSGetExecutablePath'
    auto result = _NSGetExecutablePath(buf, &bufsize);
                  ^
jiyu/src/os_support.cpp:48:29: error: no viable conversion from returned value of type
      'const char [1]' to function return type 'String'
    if (result != 0) return "";
                            ^~
jiyu/src/general.h:29:8: note: candidate constructor (the implicit copy constructor) not viable:
      no known conversion from 'const char [1]' to 'const String &' for 1st argument
struct String {
       ^
jiyu/src/general.h:29:8: note: candidate constructor (the implicit move constructor) not viable:
      no known conversion from 'const char [1]' to 'String &&' for 1st argument
jiyu/src/os_support.cpp:61:30: error: use of undeclared identifier 'F_OK'
        bool result = access(c_str, F_OK) != -1;
                                    ^
3 errors generated.
make[2]: *** [CMakeFiles/jiyu.dir/src/os_support.cpp.o] Error 1
make[1]: *** [CMakeFiles/jiyu.dir/all] Error 2
make: *** [all] Error 2
```
```
$ clang --version
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Also, I'm using the binary `clang+llvm-8.0.0-x86_64-apple-darwin` downloaded from LLVM's official website instead of building from source; I don't think that would cause these errors though.

--- a/src/os_support.cpp
+++ b/src/os_support.cpp
@@ -38,6 +38,7 @@ bool file_exists(String path) {
 #endif
 
 #ifdef MACOSX
+#include <mach-o/dyld.h>
 
 String get_executable_path() {
     const u32 BUFFER_SIZE = 512;
@@ -45,7 +46,7 @@ String get_executable_path() {
 
     u32 bufsize = BUFFER_SIZE;
     auto result = _NSGetExecutablePath(buf, &bufsize);
-    if (result != 0) return "";
+    if (result != 0) return to_string("");
 
     return copy_string(to_string(buf));
 }
@@ -54,6 +55,7 @@ String get_executable_path() {
 // @Incomplete get_executable_path for Linux
 
 #ifdef UNIX
+#include <unistd.h>
 #include <sys/stat.h>
 
 bool file_exists(String path) {

